Chatkit/Chatkit/Chatkit.swift

```diff
@@ -56,16 +69,41 @@ public class Chatkit {
     ///     - completionHandler: An optional completion handler called when a connection has
     ///     been successfuly established or failed due to an error.
     public func connect(completionHandler: CompletionHandler? = nil) {
-        // TODO: Implement
-        self.connectionStatus = .connected
-        if let completionHandler = completionHandler {
-            completionHandler(nil)
+        
+        // TODO: Implement properly
+        
+        switch connectionStatus {
+
+        case .disconnected:
+            
+            dependencies.subscriptionManager.subscribe(toType: .user, sender: self) { result in
+                switch result {
+                case .success:
+                    self.connectionStatus = .connected
+                    completionHandler?(nil)
```

Comment:
Good suggestion(s).  I have updated the code so that the `completionHandler` is always invoked on the main thread. 

Question for you...

'm having to unwrap the optional `completionHandler` all over the place (in order to avoid an unncessary loop on the main thread in the event of a nil). This is rather verbose and a feels a little ugly. 

A possible alternative would be to introduce a convenience method.  Perhaps something like: 

```
func execute<T>(_ closure: ((T) -> Void)?, onMainThreadWith argument: T) {
    guard let closure = closure else {
        return
    }
    
    DispatchQueue.main.async {
        closure(argument)
    }
}
```

which can then be invoked like so: 
```
let error = ChatkitError.connecting
execute(completion, onMainThreadWith: error)
```

I appreciate this is pretty subjective so I wanted to get your opinion on this before implementing.  Are thoughts?

Reply:
Sure, I think we can do it like this.